### ISSUE TITLE
`@stylistic/type-annotation-spacing`: Confused by parentheses

### ISSUE THREAD

**Issue by ej612** [eslint-stylistic/eslint-stylistic#441]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Hi there!

This is my eslint config:
```
'@stylistic/type-annotation-spacing': ['error', {
    overrides: {
        colon: {
            before: false,
            after: true
        }
    }
}]
```

And this is my code:
```ts
/**
 * Missing space between colon & return type is detected
 * -> GOOD
 */
function error_is_reported_ok():void {
    //
}
```

An error is displayed as expected:
![image](https://github.com/eslint-stylistic/eslint-stylistic/assets/110408767/f849f461-578e-4fbd-8459-337501d5285c)

If I wrap the return type in parentheses, like so:
```ts
/**
 * If return type is wrapped in parentheses,
 * missing space after colon is not detected
 * -> BAD
 */
function error_is_not_reported_not_ok():(void) {
    //
}
```

No violation is detected anymore. Is this intended behavior? I would expect the rule to report a violation, parentheses or not.

Thanks a lot in advance!

### Reproduction

https://stackblitz.com/edit/github-dgu1fv-ymqf1g

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
